### PR TITLE
Detect vertex format + vertex color + tests

### PR DESCRIPTION
**Pull request by einarf**
Change the face reader to detect vertex format and adapt to reading this format

### CODE REVIEW

**Comment by einarf** [comment]
Done and ready for scrutiny 😄

**Comment by einarf** [comment]
Test obj with only positions and uvs (normals calcualted on gpu)

test

File: pywavefront/obj.py [hunk 1]
```diff
@@ -41,12 +43,24 @@ def consume_vertices(self):
...
         while True:
-            # TODO: Check for vertex color
-            yield (
-                float(self.values[1]),
-                float(self.values[2]),
-                float(self.values[3]),
-            )
+            # Vertex color
+            if len(self.values) == 7:
+                print(self.line)
```
Comment (greenmoss) [inline pywavefront/obj.py]: Do we need this `print` line?

File: pywavefront/obj.py [hunk 2]
```diff
@@ -157,53 +172,103 @@ def consume_faces(self):
 
         We always triangulate to make it simple
         """
+        # Figure out the format of the first vertex
+        # We assume every consecutive vertex has the same format
```
Comment (greenmoss) [inline pywavefront/obj.py]: Alternate wording, based on my reading of the code below:

`We raise an exception if any following vertex has a different format`

File: pywavefront/obj.py [hunk 3]
```diff
@@ -157,53 +172,103 @@ def consume_faces(self):
...
+            v1, vlast = None, None
 
             for i, v in enumerate(self.values[1:]):
-                v_index, t_index, n_index = (list(map(int, [j or 0 for j in v.split('/')])) + [0, 0])[:3]
+                parts = v.split('/')
+                v_index = (int(parts[0]) - 1)
+                t_index = (int(parts[1]) - 1) if has_vt else None
+                n_index = (int(parts[2]) - 1) if has_vn else None
+
+                # v_index, t_index, n_index = (list(map(int, [j or 0 for j in v.split('/')])) + [0, 0])[:3]
```
Comment (greenmoss) [inline pywavefront/obj.py]: We can remove this comment, right?

File: pywavefront/obj.py (hunk 1 above)
Comment (einarf) [reply pywavefront/obj.py]: Definitely not!

File: pywavefront/obj.py (hunk 3 above)
Comment (einarf) [reply pywavefront/obj.py]: fixed

File: pywavefront/obj.py (hunk 2 above)
Comment (einarf) [reply pywavefront/obj.py]: fixed

**Comment by greenmoss** [comment]
Do you need a release now, or do you have a few more? I can go either way.

**Comment by einarf** [comment]
No hurry for me personally. Could be an idea to update `visialization` module to be able to draw these new formats first. Also `README` could need some love 😄

**Comment by greenmoss** [comment]
Kewl; I'll hold off.

**Pull request merged**